test/controllers/CategoryGuidanceControllerSpec.scala

```diff
@@ -17,17 +17,92 @@
 package controllers
 
 import base.SpecBase
+import connectors.OttConnector
+import models.Commodity
+import models.ott.response.{GoodsNomenclatureResponse, OttResponse}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{times, verify, when}
+import org.scalatestplus.mockito.MockitoSugar.mock
+import play.api.inject._
 import play.api.test.FakeRequest
 import play.api.test.Helpers._
+import queries.CommodityQuery
+import repositories.SessionRepository
 import views.html.CategoryGuidanceView
 
+import scala.concurrent.Future
+
 class CategoryGuidanceControllerSpec extends SpecBase {
 
   "CategoryGuidance Controller" - {
 
+    val userAnswersWithCommodity = emptyUserAnswers
+      .set(
+        CommodityQuery,
+        Commodity(commodityCode = "123", description = "test commodity")
+      )
+      .success
+      .value
+
+    val mockOttConnector = mock[OttConnector]
+    when(mockOttConnector.getCategorisationInfo(any())(any())).thenReturn(
+      Future.successful(
+        OttResponse(
+          GoodsNomenclatureResponse("", ""),
+          Seq(),
+          Seq()
+        )
+      )
+    )
+
+    "must call OTT and save the response in user answers prior to loading on a GET" in {
+
+      val mockSessionRepository = mock[SessionRepository]
+      when(mockSessionRepository.set(any())).thenReturn(Future.successful(true))
+
+      val application = applicationBuilder(userAnswers = Some(userAnswersWithCommodity))
+        .overrides(
+          bind[OttConnector].toInstance(mockOttConnector),
+          bind[SessionRepository].toInstance(mockSessionRepository)
+        )
+        .build()
+
+      running(application) {
+
+        val request = FakeRequest(GET, routes.CategoryGuidanceController.onPageLoad().url)
+        val result  = route(application, request).value
+
+        status(result) mustEqual OK
+        verify(mockOttConnector, times(1)).getCategorisationInfo(any())(any())
+        verify(mockSessionRepository, times(1)).set(any())
+
+      }
+    }
+
+    "must redirect to Journey Recover when no commodity query has been provided" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
+        .overrides(
+          bind[OttConnector].toInstance(mockOttConnector)
+        )
+        .build()
+
+      running(application) {
+        val request = FakeRequest(GET, routes.CategoryGuidanceController.onPageLoad().url)
+        val result  = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual routes.JourneyRecoveryController.onPageLoad().url
```

Comment:
Could check in here that we don't call the ottConnector if there is no commodity query? Might be unnecessary here, just popped into mind since we are binding the connector anyway

Reply:
Done :)